browser/resources/brave_new_tab_page_refresh/components/search/search_box.tsx

```diff
@@ -142,19 +142,19 @@ export function SearchBox(props: Props) {
 
   function updateSelectedOption(step: number) {
     if (resultOptions.length === 0) {
-      setSelectedOption(optional())
+      setSelectedOption(null)
       return
     }
-    let index = selectedOption.valueOr(-1) + step
-    if (!selectedOption.hasValue() && step <= 0) {
+    let index = (selectedOption ?? -1) + step
+    if (selectedOption === null && step <= 0) {
```

Comment:
Can this be simplified to 
```suggestion
    if (step <= 0) {
```

depends on the range of step, I guess (if its `-1` <==> `1`)